core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -107,10 +107,14 @@ public static Schema buildAvroProjection(Schema schema, org.apache.iceberg.Schem
 
   public static boolean isTimestamptz(Schema schema) {
     LogicalType logicalType = schema.getLogicalType();
-    if (logicalType != null && logicalType instanceof LogicalTypes.TimestampMicros) {
+    if (logicalType instanceof LogicalTypes.TimestampMillis || logicalType instanceof LogicalTypes.TimestampMicros) {
       // timestamptz is adjusted to UTC
       Object value = schema.getObjectProp(ADJUST_TO_UTC_PROP);
-      if (value instanceof Boolean) {
+
+      if (value == null) {
+        // not all avro timestamp logical types will have the adjust_to_utc prop, default to not timestamptz
+        return false;
```

Comment:
This is the same behavior as before because `null` would not pass either `instanceof` check. Do we need to update this other than to add `instanceof TimestampMillis` above?

Reply:
The instanceof check is on the *logical type* not the object prop. So the instanceof check will pass and then the object prop will be null and hit this branch.

Reply:
WRT adding `instanceof TimestampMillis` that was an unrelated bug it seems. 

There were two places a similar check was done `AvroSchemsUtil.isTimestamptz` and `SchemaToType.primitive` but the two places were not consistent. I am just making those two checks both use the same utility function.

Reply:
You are totally right this change is not strictly necessary, just makes it more explicit. 

The bit that is important is the change in this file which before failed: `core/src/main/java/org/apache/iceberg/avro/SchemaToType.java`